spec/amber/support/params_spec.cr

```diff
@@ -0,0 +1,43 @@
+require "../../../spec_helper"
+require "http"
+
+module Amber
+    module Support
+        describe Params do
+
+            describe "#validate" do
+                it "validates required field" do
+                    http_params = HTTP::Params.parse("name=elias&last_name=perez")
+                    params = Params.new(http_params)
+
+                    params.validate do
+                        required("name") { |v| v.str? & !v.empty? }
+                        required("last_name") { |v| v.str? & !v.empty? }
+                    end
+
+                    params.valid?.should eq true
+                end
+            end
+
+            describe "#required" do
+                it "returns true for all predicates" do
+                    http_params = HTTP::Params.parse("name=elias&last_name=perez")
+
+                    params = Params.new(http_params)
+                    result = params.required("name") { |v| v.str? & !v.empty? & !v.email? }
+
+                    result.should eq true
+                end
+
+                it "returns false if one predicate is false" do
+                    http_params = HTTP::Params.parse("name=elias&last_name=perez")
+
+                    params = Params.new(http_params)
+                    result = params.required("name") { |v| v.str? & !v.empty? & v.email? }
+
+                    result.should eq false
+                end
+            end
+        end
+    end
+end
```

Comment:
yes, I will reformat still in progress :) I just wanted to show you guys the Amber::Controller::Base file